---
#Title, authors, date, doi:
title: "Sparse canonical methods for biological data integration: application to a cross-platform study."
authors: ["Cao Le", "PG Martin", "C Robert-Granie", "P Besse"]
date: 2009-03-17
doi: ""

# Schedule page publish date (NOT publication date).
publishDate: 2020-08-30T19:43:42-04:00

# Publication type
# Legend: 0=uncategorized, 1=conf paper, 2=article, 3=preprint, 4=report, 5=book, 6=book chapter, 7=thesis, 8=patent
publication_types: ["2"]

# Publication name and optional abbreviated name
publication: "BMC bioinformatics"
publication_short: "BMC Bioinformatics. 2009 Jan 26;10:34."

# Abstract:
abstract: "BACKGROUND: In the context of systems biology, few sparse approaches have been proposed so far to integrate several data sets. It is however an important and fundamental issue that will be widely encountered in post genomic studies, when simultaneously analyzing transcriptomics, proteomics and metabolomics data using different platforms, so as to understand the mutual interactions between the different data sets. In this high dimensional setting, variable selection is crucial to give interpretable results. We focus on a sparse Partial Least Squares approach (sPLS) to handle two-block data sets, where the relationship between the two types of variables is known to be symmetric. Sparse PLS has been developed either for a regression or a canonical correlation framework and includes a built-in procedure to select variables while integrating data. To illustrate the canonical mode approach, we analyzed the NCI60 data sets, where two different platforms (cDNA and Affymetrix chips) were used to study the transcriptome of sixty cancer cell lines. RESULTS: We compare the results obtained with two other sparse or related canonical correlation approaches: CCA with Elastic Net penalization (CCA-EN) and Co-Inertia Analysis (CIA). The latter does not include a built-in procedure for variable selection and requires a two-step analysis. We stress the lack of statistical criteria to evaluate canonical correlation methods, which makes biological interpretation absolutely necessary to compare the different gene selections. We also propose comprehensive graphical representations of both samples and variables to facilitate the interpretation of the results. CONCLUSION: sPLS and CCA-EN selected highly relevant genes and complementary findings from the two data sets, which enabled a detailed understanding of the molecular characteristics of several groups of cell lines. These two approaches were found to bring similar results, although they highlighted the same phenomenons with a different priority. They outperformed CIA that tended to select redundant information."

# Optional shortened summary:
summary: ""

# Tags and categories:
tags: []
categories: []

# Featured/Selected article?
featured: false

# Custom links
links:
- name: Pubmed
  url: https://www.ncbi.nlm.nih.gov/pubmed/19171069
  icon_pack: ai
  icon: pubmed

# Links
url_pdf:
url_code:
url_dataset:
url_poster:
url_project:
url_slides:
url_source:
url_video:

# Featured image (add jpg/png to page folder)
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight
image: 
  caption: "19171069"
  focal_point: ""
  preview_only: ""

# Associated projects (myproject references content/project/myproject/index.md)
projects: [GeneExpressionRegulation]

# Slides (slides: "myslides" references content/slides/myslides/index.md)
slides: ""
---
