---
# Documentation: https://sourcethemes.com/academic/docs/managing-content/

title: "Nuclear receptors"
summary: "Regulation of gene expression in the liver by nuclear receptors"
authors: []
tags: ["genomics", "gene expression", "transcription", "liver", "lipid metabolism"]
categories: ["gene expression", "genomics"]
date: 2020-08-30T22:57:36-04:00

# Optional external URL for project (replaces project detail page).
external_link: ""

# Featured image
# To use, add an image named `featured.jpg/png` to your page's folder.
# Focal points: Smart, Center, TopLeft, Top, TopRight, Left, Right, BottomLeft, Bottom, BottomRight.
image:
  caption: "PPAR-RXR"
  focal_point: "Center"
  preview_only: true

# Custom links (optional).
#   Uncomment and edit lines below to show custom links.
# links:
# - name: Follow
#   url: https://twitter.com
#   icon_pack: fab
#   icon: twitter

url_code: ""
url_pdf: ""
url_slides: ""
url_video: ""

# Slides (optional).
#   Associate this project with Markdown slides.
#   Simply enter your slide deck's filename without extension.
#   E.g. `slides = "example-slides"` references `content/slides/example-slides.md`.
#   Otherwise, set `slides = ""`.
slides: ""

# Optional header
header:
  image: "headers/PPAR-RXR_bandeau.png"
  caption: "PPAR-RXR"

---

Nuclear receptors (NR) are *ligand-activated transcription factors* that regulate numerous biological processes in metazoans. The human genome encodes 48 NRs (49 in mouse, 21 in Drosophila and >270 in Caenorabditis).   

Nuclear receptors share a common structure and are organized in independent functional domains.   

{{< figure library="true" src="Fig_NucRec_Structure.png" title="Nuclear Receptor Structure" >}}  

The DNA binding domain (DBD) is the most conserved domain. The structure of the ligand binding domain (LBD), which defines a ligand binding pocket buried in a 3 layer "alpha-helical sandwich" is also well conserved among the nuclear receptors. The conformational changes induced upon ligand binding modifies the interactions with various proteins, including cofactors involved in chromatin remodeling, nucleosome modification and interactions with the basal transcription machinery.   

Nuclear receptors regulate gene expression as monomers, homodimers or heterodimers, which contributes to define the type of DNA response element they can bind to in the promoters of their target genes. Traditionnally 4 classes of nuclear receptors have been defined:
  
{{< figure library="true" src="Fig_NucRec_Classes.png" title="Classes of Nuclear Receptors" >}}  
  

* Steroid hormone receptors (androgen receptor AR, estrogen receptors ER, glucocorticoid receptor GR, mineralocorticoid receptor MR and progesterone receptor PR) comprise the **class I nuclear receptors**. These receptors function as homodimers, are activated by low doses of steroid hormones, bind IR3-like response elements (inverted repeats separated by 3 nucleotides) and regulate numerous physiological processes, including reproduction, metabolism or immunity.

* **Class II** is composed of the nuclear receptors that function as heterodimers with the RXR (Retinoïd X Receptors). They include the PPARs, LXRs, FXR, RAR, PXR, CAR, VDR and TR. They generally bind to DR-like response elements (direct repeats of the consensus motif AGGTCA separated by a variable number of nucleotides) but other types of response elements have also been described.

* **Class III** members are the non-steroid hormone nuclear receptors that also function as homodimers such as RXRs, HNF4 or COUP-TF and typically bind DR-like motifs.

* Nuclear receptors of **Class IV** function as monomers and bind a single consensus motif. This class includes NGFI_B, Nur1, SF-1, LRH-1 and RORs.

*Note that some receptors might function as monomer or as dimers, such as rev-erb NRs.*  
  
During the first ~12 years of my career, I have mostly focused on better understanding the role played by the class II nuclear receptors PPARalpha annd RXR on the regulation of gene expression in the liver.  
  
The liver plays central roles in the detoxification of xenobiotics, including drugs and toxicants, but also in energy metabolism, notably via its roles in lipid and lipoprotein metabolism. As such, it represents both a relevant drug target and a tissue which dysfunction can have broad, systemic consequences.   
  
During my PhD, I studied the activation of the PPARalpha-RXR heterodimer through **nutritional** (dietary fatty acids; [Martin *et al.*, Hepatology, 2007]({{< relref "publication/2007-03-26_17326203/index.md" >}})), **pharmacological** (RXR and PPARalpha activating-drugs; [Martin *et al.*, Gene Expr, 2006]({{< relref "publication/2006-06-12_16128002/index.md" >}})) and **physio-pathological** cues (fasting; [Déjean *et al.*, 2010]({{< relref "publication/2010-06-28_17713590/index.md" >}}), [Blavy *et al.*, 2009]({{< relref "publication/2011-02-24_19635486/index.md" >}})) and the corresponding impacts on gene expression and lipid storage in the liver (hepatic steatosis). See my {{% staticref "files/PhD_manuscript_Pascal_MARTIN.pdf" %}}PhD manuscript{{% /staticref %}} for further details.  
  
Later, I shared my experience in the analysis of the transcriptome by qPCR and microarrays by developping a transcriptomic facility in my research department: the [GeT-TRiX facility](https://www6.toulouse.inra.fr/toxalim/Plateformes-Technologiques/E23-TRiX). With the development of the [ToxAlim](https://www6.toulouse.inra.fr/toxalim) INRA unit, I went on to address research questions in molecular [food toxicology]({{< relref "project/foodtoxicology/index.md" >}}).    

